lib/win32.js

```diff
@@ -17,17 +17,35 @@
 'use strict';
 
 const Promise = require('bluebird');
+const tmp = Promise.promisifyAll(require('tmp'));
+const fs = Promise.promisifyAll(require('fs'));
+const debug = require('debug')(require('../package.json').name);
 const retry = require('bluebird-retry');
+const childProcess = Promise.promisifyAll(require('child_process'));
 const os = require('os');
 const _ = require('lodash');
 
+tmp.setGracefulCleanup();
+debug.log = console.log.bind(console);
+
 const runDiskpartScript = (script) => {
-  return Promise.try(() => {
-    if (os.platform() === 'win32') {
-      const diskpart = Promise.promisifyAll(require('diskpart'));
-      return diskpart.evaluateAsync(script).delay(2000);
-    }
-  });
+  if (os.platform() !== 'win32') {
+    return Promise.resolve();
+  }
+
+  return tmp.fileAsync().tap((temporaryPath) => {
+    return fs.writeFileAsync(temporaryPath, _.join(script, '\n'));
+  }).then((temporaryPath) => {
+    return childProcess.execAsync(`diskpart /s "${temporaryPath}"`);
```

Comment:
I guess it'll be great when we finally have native locking because then we can get rid of these kludgy `temporaryPath` files too :)